chain/epoch_manager/src/lib.rs

```diff
@@ -56,6 +59,8 @@ pub struct EpochManager {
     epoch_info_aggregator: Option<EpochInfoAggregator>,
     /// Largest final height. Monotonically increasing.
     largest_final_height: BlockHeight,
+    /// Cache of compiled to native code contracts.
+    cache: Arc<dyn CompiledContractCache>,
```

Comment:
it makes no sense to have the contract cache here

Reply:
> 1. Please add description.
You mean to PR itself?

> 2. Why contract cache belongs to epoch manager? It doesn't make any sense. I don't think this hack is a good idea.
OK, actually I agree that it's not the best place, however feel that view call miss smth like `ApplyState`, and adding additional argument felt somewhat clumsy. WDYT of replacing `epoch_info_provider` with more general purpose argument of that kind?

> 3. `near-vm-runner` can't depend on primitives, because we'll not be able to publish it to crates.
Hmm, it already depends on `near_primitives::hash::CryptoHash` per your suggestion in previous review. How would you suggest to organize dependency graph?

Reply:
EpochInfoProvider is supposed to provide specifically information related to epoch and validators and we definitely should not tag on the contract cache there.